source/adios2/helper/adiosNetwork.cpp

```diff
@@ -18,11 +18,12 @@
 #include <iostream>
 #include <thread>
 
-#include <arpa/inet.h> //AvailableIpAddresses() inet_ntoa
-#include <net/if.h>    //AvailableIpAddresses() struct if_nameindex
-#include <string.h>    //AvailableIpAddresses() strncp
-#include <sys/ioctl.h> //AvailableIpAddresses() ioctl
-#include <unistd.h>    //AvailableIpAddresses() close
+#include <arpa/inet.h>  //AvailableIpAddresses() inet_ntoa
+#include <net/if.h>     //AvailableIpAddresses() struct if_nameindex
```

Comment:
I have tried to make it compile on FreeBSD very long time ago, but as far as I remember these socket header files did not exist on BSD systems. Just wondering if there is any tricks I didn't know, or do they actually exist on BSD?

Reply:
Side note: I didn't remove any include statements, I only changed the indentation of the comments, which is shown as both removal and addition of new lines.

I compiled the code for FreeBSD in a Docker container via [BinaryBuilder](https://binarybuilder.org). I did not set up the container myself, but I assume it's reasonably standard. I don't know about other BSDs, or about native BSD systems.